docs/references/tanstack-start/get-auth.mdx

```diff
@@ -0,0 +1,117 @@
+---
+title: '`getAuth()`'
+description: The getAuth() helper retrieves the authentication state allowing you to protect your API routes or gather relevant data.
+---
+
+The `getAuth()` helper retrieves the authentication state, allowing you to protect your routes or gather relevant data.
+
+## Parameters
+
+<Properties>
+  - `request`
+
+  The Request object.
+
+  ---
+
+  - `opts?`
+
+  An optional object that can be used to configure the behavior of the `getAuth()` function. It accepts the following properties:
+
+  - `secretKey?`: A string that represents the secret key used to sign the session token. If not provided, the secret key is retrieved from the environment variable `CLERK_SECRET_KEY`.
+</Properties>
+
+## Returns
+
+`getAuth()` returns the [`Auth`](/docs/references/nextjs/auth-object) object.
+
+## Usage
+
+### Basic usage
+
+The following example demonstrates how to use `getAuth()` to retrieve authentication information in a server function and API routes.
```

Comment:
This specific example only shows how to use getAuth in a server function - not also an API route. Let's remove that part from the sentence.

```suggestion
The following example demonstrates how to use `getAuth()` to retrieve authentication information in a server function.
```